src/ray/core_worker/core_worker.cc

```diff
@@ -2963,8 +2968,33 @@ Status CoreWorker::SealReturnObject(const ObjectID &return_id,
   return status;
 }
 
-void CoreWorker::DelObjectRefStream(const ObjectID &generator_id) {
-  task_manager_->DelObjectRefStream(generator_id);
+void CoreWorker::AsyncDelObjectRefStream(const ObjectID &generator_id) {
+  RAY_LOG(DEBUG) << "AsyncDelObjectRefStream " << generator_id;
+  deleted_generator_ids_.insert(generator_id);
+}
+
+void CoreWorker::TryDeleteObjectRefStreams() {
+  std::vector<ObjectID> out_of_scope_generator_ids;
+  for (auto it = deleted_generator_ids_.begin(); it != deleted_generator_ids_.end();
+       it++) {
+    const auto &generator_id = *it;
+    RAY_LOG(DEBUG) << "Try DelObjectRefStream " << generator_id;
+    int64_t num_objects_generated = 0;
+    if (!task_manager_->StreamingGeneratorIsFinished(generator_id,
+                                                     &num_objects_generated)) {
+      continue;
+    }
+
+    bool can_gc = reference_counter_->CheckGeneratorRefsOutOfScope(generator_id,
```

Comment:
can we remove all unconsumed objects here (and add a test)? This seems like a regression (where unconsumed objects are not deleted if lineage is alive). 

E.g., when delete hapapens, all unconsumed objects are deleted (same behavior as now). And only the stream metadata is cleaned up after a delay.